### PR TITLE
Gfx 10.2 caps and deactivate GDI unsupported functions

### PR DESCRIPTION
**Pull request by akallabeth**
* Added GFX 10.2 capability
* Gracefully handling unknown capabilities in GFX server channel
* Deactivated PolygonSC and PolygonCB in capabilities of X11 and Wayland clients as it is not implemented.

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test please

**Comment by realjiangms** [comment]
I just pulled your branch and tested with my win10. It works perfectly.

File: channels/rdpgfx/server/rdpgfx_main.c [hunk 1]
```diff
@@ -345,8 +345,8 @@ static UINT rdpgfx_send_create_surface_pdu(RdpgfxServerContext* context,
  *
  * @return 0 on success, otherwise a Win32 error code
  */
-UINT rdpgfx_send_delete_surface_pdu(RdpgfxServerContext* context,
-                                    RDPGFX_DELETE_SURFACE_PDU* pdu)
+static UINT rdpgfx_send_delete_surface_pdu(RdpgfxServerContext* context,
+        RDPGFX_DELETE_SURFACE_PDU* pdu)
```
Comment (realjiangms) [inline channels/rdpgfx/server/rdpgfx_main.c]: I noticed this format change. Why it looks different rule from rdpgfx_send_solid_fill_pdu below?

File: channels/rdpgfx/server/rdpgfx_main.c [hunk 2]
```diff
@@ -1170,31 +1170,30 @@ static UINT rdpgfx_recv_caps_advertise_pdu(RdpgfxServerContext* context,
...
+
+	if (!capsSets)
+		return ERROR_OUTOFMEMORY;
+
 	pdu.capsSets = (RDPGFX_CAPSET*) capsSets;
 
 	for (index = 0; index < pdu.capsSetCount; index++)
 	{
-		capsSet = &(pdu.capsSets[index]);
+		RDPGFX_CAPSET* capsSet = &(pdu.capsSets[index]);
```
Comment (realjiangms) [inline channels/rdpgfx/server/rdpgfx_main.c]: I notice the declare of capsSet is moved into this code block. Some compiler might ask everything to be declared at very beginning of a function for .c files. If we don't want to be compatible with such compiler, then it's just OK.

**Comment by realjiangms** [comment]
For reference, it fixes #3559

File: channels/rdpgfx/server/rdpgfx_main.c (hunk 2 above)
Comment (akallabeth) [inline channels/rdpgfx/server/rdpgfx_main.c]: @realjiangms Are you sure? Block scope should be compatible with C89

Comment (hardening) [inline channels/rdpgfx/server/rdpgfx_main.c]: Yes definitely

Comment (akallabeth) [inline channels/rdpgfx/server/rdpgfx_main.c]: @hardening this means compatible or not?

Comment (mfleisz) [inline channels/rdpgfx/server/rdpgfx_main.c]: Yes - compatible I'd say!

Comment (hardening) [inline channels/rdpgfx/server/rdpgfx_main.c]: Yes compatible of course

File: include/freerdp/channels/rdpgfx.h [hunk 3]
```diff
@@ -92,7 +92,9 @@ typedef struct _RDPGFX_HEADER RDPGFX_HEADER;
 #define RDPGFX_CAPVERSION_8			0x00080004
 #define RDPGFX_CAPVERSION_81			0x00080105
 #define RDPGFX_CAPVERSION_10			0x000A0002
+#define RDPGFX_CAPVERSION_102			0x000A0200
```
Comment (bmiklautz) [inline include/freerdp/channels/rdpgfx.h]: Would be great if we can add some reference about the new cap.

**Comment by akallabeth** [comment]
@freerdp-bot Don't be like that, test!

**Comment by akallabeth** [comment]
@mfleisz @hardening @bmiklautz Anyone interested in merging?

File: channels/rdpgfx/server/rdpgfx_main.c (hunk 2 above)
Comment (realjiangms) [inline channels/rdpgfx/server/rdpgfx_main.c]: Yes it's compatible. Long time ago when I failed, I probably didn't define it at beginning of a block scope

**Pull request merged**